Sources/Client/Model/Message+Requests.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  Message+Requests.swift
+//  StreamChatCore
+//
+//  Created by Alexey Bukhtin on 10/01/2020.
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+public extension Message {
+    
+    /// Delete the message.
+    /// - Parameter completion: a completion block with `MessageResponse`.
+    @discardableResult
+    func delete(_ completion: @escaping Client.Completion<MessageResponse>) -> URLSessionTask {
+        return Client.shared.request(endpoint: .deleteMessage(self), completion)
+    }
+    
+    /// Send a request for reply messages.
+    /// - Parameters:
+    ///   - pagination: a pagination (see `Pagination`).
+    ///   - completion: a completion block with `[Message]`.
+    @discardableResult
+    func replies(pagination: Pagination, _ completion: @escaping Client.Completion<[Message]>) -> URLSessionTask {
+        let completion = doAfter(completion) { messages in
+            self.add(repliesToDatabase: messages)
+        }
+        
+        return Client.shared.request(endpoint: .replies(self, pagination)) { (result: Result<MessagesResponse, ClientError>) in
+            completion(result.map({ $0.messages }))
+        }
+    }
+    
+    // MARK: - Reactions
+    
+    /// Add a reaction to the message.
+    /// - Parameters:
+    ///   - reactionType: a reaction type, e.g. like.
+    ///   - completion: a completion block with `MessageResponse`.
+    @discardableResult
+    func addReaction(_ reactionType: ReactionType, _ completion: @escaping Client.Completion<MessageResponse>) -> URLSessionTask {
+        return Client.shared.request(endpoint: .addReaction(reactionType, self), completion)
+    }
+    
+    /// Delete a reaction to the message.
+    /// - Parameters:
+    ///   - reactionType: a reaction type, e.g. like.
+    ///   - completion: a completion block with `MessageResponse`.
+    @discardableResult
+    func deleteReaction(_ reactionType: ReactionType,
+                        _ completion: @escaping Client.Completion<MessageResponse>) -> URLSessionTask {
+        return Client.shared.request(endpoint: .deleteReaction(reactionType, self), completion)
+    }
+    
+    // MARK: Flag Message
+    
+    /// Flag a message.
+    /// - Parameter completion: a completion block with `FlagMessageResponse`.
+    @discardableResult
+    func flag(_ completion: @escaping Client.Completion<FlagMessageResponse>) -> URLSessionTask {
+        if user.isCurrent {
```

Comment:
Are we going to remove these checks so server handles them?

Reply:
I think we should keep it as it now.